main/src/ca/uwaterloo/flix/language/phase/jvm/GenExpression.scala

```diff
@@ -1121,7 +1121,7 @@ object GenExpression {
           }
           // Calling unwind and unboxing
 
-          if (purity == Purity.Pure) BackendObjType.Result.unwindSuspensionFreeThunk("in pure closure call", loc)(new BytecodeInstructions.F(mv))
+          if (!Purity.canUseAlgebraicEffects(purity)) BackendObjType.Result.unwindSuspensionFreeThunk("in pure closure call", loc)(new BytecodeInstructions.F(mv))
```

Comment:
Would it not be easier if it was called "isPure" here?

Reply:
In the next PR, both Pure and Impure should take this branch. So I thought the most direct and future-proof phrasing would be `canUseAlgebraicEffects`.

Reply:
Yes